get-artifacts/get_artifacts.sh

```diff
@@ -0,0 +1,168 @@
+#!/bin/bash
+function echoerr() {
+  cat <<< "$@" 1>&2;
+}
+
+#Check prerequists.
+function check_command() {
+    for var in "$@"; do
+      if ! (command -v "$var" >/dev/null 2>&1); then
+          echoerr "This command requires $var to be installed"
+          exit 1
+      fi
+    done
+}
+check_command jq curl zip echo getopts
+
+function usage() {
+  echoerr -t token used to communicate with github actions.
+  echoerr -h number of workflow_runs to pull artifacts from.
+  echoerr -a space seperated list of artifact names.
+  echoerr -b optional branch to pull historical artifacts from.
+  echoerr -w workflow file name of of workflow to pull artifact from.
+  echoerr -d target directory where subdirectories for jobs will be created and artifacts will be unzip to.
+  echoerr -i the single job to artifacts from.  Overrides -b,-w, and -h
+  echoerr -z decompress and delete the original downloaded artifacts?
+  echoerr -? this message.
+  echoerr output will written in to the -t target directory.
+}
+
+TOKEN=
+HISTORY=
+ARTIFACTS=
+REPO=
+WORKFLOW=
+BRANCH=
+TARGET_DIR=
+WORKFLOW_RUN_ID=
+DECOMPRESS=false
+
+
+while getopts 'h:a:r:w:b:i:t:d:z' OPTION; do
+  case "$OPTION" in
+    h)
+      HISTORY="$OPTARG"
+      ;;
+    a)
+      IFS=' ' read -ra ARTIFACTS <<< "$OPTARG"
+      ;;
+    r)
+      REPO="$OPTARG"
+      ;;
+    w)
+      WORKFLOW="$OPTARG"
+      ;;
+    b)
+      BRANCH="$OPTARG"
+      ;;
+    i)
+      WORKFLOW_RUN_ID="$OPTARG"
+      ;;
+    d)
+      TARGET_DIR="$OPTARG"
+      ;;
+    z)
+      DECOMPRESS="true"
+      ;;
+    t)
+      TOKEN="$OPTARG"
+      ;;
+    ?)
+      usage
+      exit 1
+      ;;
+  esac
+done
+
+if [[ -z "${TARGET_DIR}" ]]; then
+  echoerr target directory must be sepecified.
+  usage
+  exit 1
+fi
+
+if [[ -z "${WORKFLOW_RUN_ID}" ]]; then
+  if [[ -z "${BRANCH}" ]] || [[ -z "${WORKFLOW}" ]]; then
+    echoerr 'You must either specify a workflow_run_id (-i), or a branch (-b) and a workflow file (-w).'
+    usage
+    exit 1
+  fi
+fi
+
+# Retry seems to add a lot of time overhead.  Need to look in to this more.
+# curl_attri=("--retry" "3")
+curl_attri=('-H' 'Accept: application/vnd.github.v3+json')
+if [ -n "$TOKEN" ]; then
+  curl_attri+=('-H' "authorization: Bearer ${TOKEN}")
+fi
+echoerr Curl Parameters: "${curl_attri[@]}"
+
+
+# Conditionally retrieves the artifacts from a single jobs, the input parameter which must correspond to a numeric workflow_run_id in github actions
+# and stores the desired artifacts, contained in a list called "ARTIFACTS", in a subdirectory of TARGET_DIR, equal to the workflow_run_id/artifact_name.
+# Artifacts are always zips.   The zip file is downloaded and optionally unziped in to the job's sub directory in the TARGET_DIR, and then deleted.
```

Comment:
Spacing, and `unzipped`. What does "job's sub directory" mean here? Should maybe be something like `TARGET_DIR/${workflow_run_id}` or similar.